### ISSUE TITLE
Skaffold dev crashes if monitored file is deleted

### ISSUE THREAD

**Issue by d11wtq** [GoogleContainerTools/skaffold#718]
If I delete a file that was added in a Dockerfile, skaffold crashes:

```
time="2018-06-21T15:01:07+10:00" level=fatal msg="statting file ../../../foo.txt: stat ../../../foo.txt: no such file or directory"
```

### Expected behavior

Deleting files should trigger a re-deploy, but should not crash skaffold.

### Actual behavior

Deleting a file that skaffold is monitoring causes it to crash.

### Information

- Skaffold version: 0.7.0
- Operating system: MacOS 10.12.16

### Steps to reproduce the behavior

1. Deploy an application with a directory `COPY`d in Dockerfile, using `skaffold dev`.
2. Delete a file from the directory that was `COPY`d into the image.
3. Skaffold will crash attempting to stat the file.

This may also happen when the file was `COPY`d directly and not via a directory. Haven't tested that.

**Comment by dgageot** [comment]
Should be fixed by https://github.com/GoogleContainerTools/skaffold/pull/620 when it's merged

**Comment by d11wtq** [comment]
Thanks @dgageot.

**Comment by dgageot** [comment]
Fixed with #620
